0707-Design-Linked_list.py

```diff
@@ -0,0 +1,132 @@
+
+Explanation:
+MyLinkedList linkedList = new MyLinkedList(); // Initialize empty LinkedList
+linkedList.addAtHead(1);
+linkedList.addAtTail(3);
+linkedList.addAtIndex(1, 2);  // linked list becomes 1->2->3
+linkedList.get(1);            // returns 2
+linkedList.deleteAtIndex(1);  // now the linked list is 1->3
+linkedList.get(1);            // returns 3
+ 
+
+Constraints:
+
+0 <= index,val <= 1000
+Please do not use the built-in LinkedList library.
+At most 2000 calls will be made to get, addAtHead, addAtTail,  addAtIndex and deleteAtIndex.
+'''
+class Node:
+    def __init__(self, data=None):
+        self.data = data
+        self.next = None
+
+class MyLinkedList:
+
+    def __init__(self):
+        """
+        Initialize your data structure here.
+        """
+        self.head = None
+        self.size = 0
+        
+
+    def get(self, index: int) -> int:
+        """
+        Get the value of the index-th node in the linked list. If the index is invalid, return -1.
+        """
+        if index < 0 or index >= self.size:
+            return -1
+        temp = self.head
+        while index > 0:
+            temp = temp.next
+            index -= 1
+        return temp.data
+        
+
+    def addAtHead(self, val: int) -> None:
+        """
+        Add a node of value val before the first element of the linked list. After the insertion, the new node will be the first node of the linked list.
+        """
+        temp = Node(val)
+        if not self.head:
+            self.head = temp
+        else:
+            temp.next = self.head
+            self.head = temp
+        self.size += 1
+        
+
+    def addAtTail(self, val: int) -> None:
+        """
+        Append a node of value val to the last element of the linked list.
+        """
+        newNode = Node(val)
+        if not self.head:
+            self.head = newNode
+        temp = self.head
+        while temp.next is not None:
+            temp = temp.next
+        temp.next = newNode
+        self.size += 1
+        
+
+    def addAtIndex(self, index: int, val: int) -> None:
+        """
+        Add a node of value val before the index-th node in the linked list. If index equals to the length of linked list, the node will be appended to the end of linked list. If index is greater than the length, the node will not be inserted.
+        """
+        if index < 0 or index > self.size:
+            return
+        if index == 0:
+            self.addAtHead(val)
+        elif index == self.size:
+            self.addAtTail(val)
+        else:
+            temp = self.head
+            i = 0
+            while i < index:
+                temp = temp.next
+                i += 1
+            temp.next = Node(val)
+            self.size += 1
+        
+
+    def deleteAtIndex(self, index: int) -> None:
+        """
+        Delete the index-th node in the linked list, if the index is valid.
+        """
+        if index < 0 or index > self.size:
```

Comment:
why using index < 0, it is clearly mentioned in the description that 0 <= index,val <= 1000